lib/chef/knife/ec2_server_create.rb

```diff
@@ -824,7 +824,7 @@ def create_server_def
           :groups => config[:security_groups],
           :security_group_ids => locate_config_value(:security_group_ids),
           :flavor_id => locate_config_value(:flavor),
-          :key_name => Chef::Config[:knife][:aws_ssh_key_id],
+          :key_name => config[:ssh_key_name] || Chef::Config[:knife][:aws_ssh_key_id],
```

Comment:
'locate_config_value' method accesses values from both the configs(CLI and knife) through the same name, but here the issue is that CLI option name in code is 'ssh_key_name', while on the other hand knife option name is 'aws_ssh_key_id', for other options the names are same so it works for them.